src/ToggleViews/ToggleViews.js

```diff
@@ -1,22 +1,87 @@
-import React, { PureComponent } from 'react';
+import React, { Component } from 'react';
 import './ToggleViews.css'
 
-export default class ToggleViews extends PureComponent {
-  onClick() {
-    //change route to page clicked on
+export default class ToggleViews extends Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      AllSessions: true,
+      LiveSessions: false,
+      MySessions: false,
+    }
+    this.onAllSessionsClick = this.onAllSessionsClick.bind(this)
+    this.onLiveSessionsClick = this.onLiveSessionsClick.bind(this)
+    this.onMySessionsClick = this.onMySessionsClick.bind(this)
+    this.filter = this.filter.bind(this)
+  }
+
```

Comment:
Sorry I forgot to Slack this to you last night, but you can combine these methods into one by passing the type of session as an argument and having the default state set as the first argument in `Object.Assign` and the changed session as the second

Reply:
It would look something like this 
`onSessionClick(session) => {
       this.sesState({
           Object.Assign({
                    AllSessions: false,
                    LiveSessions: false,
                    MySessions: false
               }, {  [session] : true})
     })
}
`